presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -327,7 +329,12 @@ public JdbcOutputTableHandle beginCreateTable(ConnectorTableMetadata tableMetada
                     fromProperties(connectionProperties));
         }
         catch (SQLException e) {
-            throw new PrestoException(JDBC_ERROR, e);
+            try {
+                dropTable(new JdbcTableHandle(connectorId, schemaTableName, catalog, schema, temporaryName));
```

Comment:
This only cleans up if the exception occurs in this method, but not during query execution. Instead, this should be done from the `rollbackCreateTable` method in `ConnectorMetadata`.